### PR TITLE
Support for YAML config format

### PR DESCRIPTION
**Pull request by premicweiss**
This is a feature based on the implementation in NestJS template. It adds support for reading YAML config. It requires YAML config in the directory, but still supports env variables via the config file, env files, which must be referenced from the YAML config. It also extracts secrets from said config.

File and variable override flow:
NestJS_config_respec

### CODE REVIEW

File: src/yargs.helper.ts [hunk 1]
```diff
@@ -56,33 +57,41 @@ export function getYargsOptions<T>(target: any): Record<keyof T, YargsOptions> {
...
+      if (["pwd", "stage", "config"].includes(name)) {
         continue;
       }
-      this[name] = values[name];
+
+      let valueFromConfig =
+        values[name] ||
+        this.config["ecs_deploy__" + name] ||
+        this.config["ecs_deploy__" + o.envAlias] ||
+        this.config[o.envAlias];
```
Comment (MarZab) [inline src/yargs.helper.ts:75]: process.env[o.envAlias]

**Comment by MarZab** [comment]
- the ecs secrets magic has been thrown out and added a simple config:

```
    ecs_secrets:
      app__test_secret: 'arn::::'
    
    ecs_env:
      app__test_fixed_env: 'test1'
```

- added unit tests, fixed env edge cases

- added a global config prefix `app__` to prevent env collisions

- upgraded node and other packages

**Review by MarZab — APPROVED** [review]
